.github/workflows/kind-e2e.yaml

```diff
@@ -0,0 +1,202 @@
+        - role: worker
+
+        # This is from: https://kubernetes.slack.com/archives/CEKK1KTN2/p1600268272383600
+        kubeadmConfigPatches:
+          - |
+            apiVersion: kubeadm.k8s.io/v1beta2
+            kind: ClusterConfiguration
+            metadata:
+              name: config
+            apiServer:
+              extraArgs:
+                "service-account-issuer": "kubernetes.default.svc"
+                "service-account-signing-key-file": "/etc/kubernetes/pki/sa.key"
+        EOF
+
+        # Create a cluster!
+        kind create cluster --config kind.yaml
+
+    - name: Install Knative Serving
+      working-directory: ./src/knative.dev/serving
+      run: |
+        set -o pipefail
+
+        kubectl create namespace knative-serving
+        kubectl apply -f config/core/300-imagecache.yaml
+
+        # Build and Publish our containers to the docker daemon (including test assets)
+        export GO111MODULE=on
+        export GOFLAGS=-mod=vendor
+        ko apply -Pf test/config/
+        ko apply -PRf config/core
+
+        # Be KinD to these tests.
+        kubectl scale -nknative-serving deployment/chaosduck --replicas=0
+
+    - name: Wait for Webhook to be up
+      working-directory: ./src/knative.dev/serving
+      run: |
+        # We need the webhook to be up
+        kubectl wait pod --for=condition=Ready -n knative-serving  -l app=webhook
+
+    - name: Install kingress provider (Contour)
+      working-directory: ./src/knative.dev/serving
+      if: matrix.kingress == 'contour'
+      run: |
+        set -o pipefail
+
+        # Apply a kingress option.
+        ko resolve -f third_party/contour-latest | \
+          sed 's/LoadBalancer/NodePort/g' | \
+          sed 's/imagePullPolicy:/# DISABLED: imagePullPolicy:/g' | \
+          kubectl apply -f -
+
+        # Have Serving use the kingress option.
+        kubectl patch configmap/config-network \
+          --namespace knative-serving \
+          --type merge \
+          --patch '{"data":{"ingress.class":"contour.ingress.networking.knative.dev"}}'
+
+        # This tells the tests what namespace to look in for our kingress LB.
+        echo "::set-env name=GATEWAY_OVERRIDE::envoy"
+        echo "::set-env name=GATEWAY_NAMESPACE_OVERRIDE::contour-external"
+
+    - name: Install kingress provider (Kourier)
+      working-directory: ./src/knative.dev/serving
+      if: matrix.kingress == 'kourier'
+      run: |
+        set -o pipefail
+
+        # Apply a kingress option.
+        ko resolve -f third_party/kourier-latest | \
+          sed 's/LoadBalancer/NodePort/g' | \
+          sed 's/imagePullPolicy:/# DISABLED: imagePullPolicy:/g' | \
+          kubectl apply -f -
+
+        # Have Serving use the kingress option.
+        kubectl patch configmap/config-network \
+          --namespace knative-serving \
+          --type merge \
+          --patch '{"data":{"ingress.class":"kourier.ingress.networking.knative.dev"}}'
```

Comment:
Seems like we could dedupe this and the above by throwing in ${{ matrix.kingress }} into the downloadlink and patch.

Reply:
Ah good call!  I added that after forgetting about these.